language/move-vm/runtime/src/session.rs

```diff
@@ -24,56 +21,52 @@ pub struct Session<'r, 'l, S> {
     pub(crate) native_extensions: NativeContextExtensions,
 }
 
-/// Result of executing a function in the VM
-pub enum ExecutionResult {
```

Comment:
I think we need to keep this type + an API that returns it around. Existing adapters are relying on it.

In addition, I think it's easier/better for an adapter to work with this structured type than it is to ask each adapter to figure out how to extract this other info (which almost every adapter will need) from other VM API's.

Totally fine with also having a more low-level API that returns `SerializedReturnValues` of course.

Reply:
This is just `execute_function` (now `execute_entry_point` function) mixed with `finish`. I don't see much of a point of keeping it around

Reply:
Can we make `finish()` return this type?

Reply:
I don't think that will have any benefit. It won't have the return values for the function or the gas used; those aren't members of the session. 

And it is already fairly descriptive on return type, so I don't think a struct wrapper buys much readability just for finish.

Reply:
I would look at the most recent fixup for a good example of the before/after with this change

Reply:
Ok, point taken. I guess I feel that writing an adapter was/is hard and this type made it a bit easier, so I'm a bit sad to see it go away. But making life easy for adapter-writers is a problem for later, no reason to block this improvement on it...